gossip-api/src/pubsub/mod.rs

```diff
@@ -26,27 +26,21 @@ pub struct AuthenticatedPubsubMessage {
 impl AuthenticatedPubsubMessage {
     /// Construct a new authenticated pubsub message from the pubsub body
     /// Sign the message if its type requires a signature
-    pub fn new_with_body(
-        body: PubsubMessage,
-        cluster_key: &SigKeypair,
-    ) -> Result<Self, SignatureError> {
+    pub fn new_with_body(body: PubsubMessage, cluster_key: &ClusterSymmetricKey) -> Self {
         // Create a signature fo the body
-        let sig = if body.requires_cluster_auth() {
-            sign_message(&body, cluster_key)?
-        } else {
-            Vec::new()
-        };
+        let sig =
+            if body.requires_cluster_auth() { create_hmac(&body, cluster_key) } else { Vec::new() };
 
-        Ok(Self { sig, body })
+        Self { sig, body }
     }
 
     /// Verify the signature on an authenticated request
-    pub fn verify_cluster_auth(&self, cluster_pubkey: &PublicKey) -> Result<(), SignatureError> {
+    pub fn verify_cluster_auth(&self, cluster_pubkey: &ClusterSymmetricKey) -> bool {
```

Comment:
tiny nit: can we call this argument `cluster_key`, or something other than `cluster_pubkey` which still implies asymmetric crypto is being used

Reply:
yes, good catch